core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2687,11 +2685,8 @@ private PhysicalOperation createLookupJoin(
             boolean spillEnabled = isSpillEnabled(session)
```

Comment:
nit: this became quite a large method. Is there a way to split it into smaller ones?

Reply:
It is not so trivial. I can obviously extract some small methods but I believe this will only hurt readability.